### PR TITLE
Replace boost::function with std::function (C++11)

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

**Review by dcousens — APPROVED** [review]

**Comment by MarcoFalke** [comment]
utACK 1b936f59264a4f4a867baece1e0ee4ec02f73cee

**Comment by sipa** [comment]
utACK 1b936f59264a4f4a867baece1e0ee4ec02f73cee

**Review by paveljanik — APPROVED** [review]
ACK

File: src/bench/bench.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include "bench.h"
 #include "perf.h"
 
+#include <assert.h>
```
Comment (practicalswift) [inline src/bench/bench.cpp]: @paveljanik That is a very good question!

Compling without `#include <assert.h>` in `src/bench/bench.cpp`:

```
  CXX      bench/bench_bench_bitcoin-bench.o
bench/bench.cpp: In member function ‘bool benchmark::State::KeepRunning()’:
bench/bench.cpp:95:86: error: ‘assert’ was not declared in this scope
     assert(count != 0 && "count == 0 => (now == 0 && beginTime == 0) => return above");
```

Compiling without `#include <limits>` in `src/bench/bench.h`:                                                                                      

```
  CXX      bench/bench_bench_bitcoin-bench_bitcoin.o
In file included from bench/bench_bitcoin.cpp:5:0:
bench/bench.h: In constructor ‘benchmark::State::State(std::__cxx11::string, double)’:
bench/bench.h:52:23: error: ‘numeric_limits’ is not a member of ‘std’
             minTime = std::numeric_limits<double>::max();
                       ^
```

Seems like `#include <boost/function.hpp>` gives you `assert.h` and `<limits>` too?

Comment (paveljanik) [inline src/bench/bench.cpp]: Perfect answer, thanks! I should have tested it myself 8)

**Comment by fanquake** [comment]
utACK 1b936f5

**Pull request merged**